src/main/java/org/example/Hobby.java

```diff
@@ -0,0 +1,56 @@
+package org.example;
+
+import java.util.ArrayList;
+
+public class Hobby {
+    private String nameOfHobby;
+    private int frequencyPerWeek;
+    ArrayList<Address> addressesForHobby = new ArrayList<>();
```

Comment:
address is enough since we already are inside a hobby object. a get now looks like this: hobby1.addressesForHobby instead of hobby1.address. Also prefer the more generic List interface on the left of the assigment: List<Address> addresses = ...